conan/tools/gnu/pkgconfigdeps.py

```diff
@@ -184,11 +184,18 @@ def shortened_content(self, info):
 
 class PCGenerator:
 
-    def __init__(self, conanfile, dep):
+    def __init__(self, conanfile, dep, build_context_suffix=None):
         self._conanfile = conanfile
+        self._build_context_suffix = build_context_suffix or {}
         self._dep = dep
+        self._dep_suffix = self._build_require_suffix(dep)
         self._content_generator = PCContentGenerator(self._conanfile, self._dep)
 
+    def _build_require_suffix(self, req):
```

Comment:
This looks a bit messy, difficult to understand, with ``build_context_suffix``, ``build_require_suffix`` and ``dep_suffix``. But if it works, probably can be left now as-is, and improved later